app/models/resource.rb

```diff
@@ -45,10 +45,10 @@ class Resource < ActiveRecord::Base
       attributesForFaceting %i[categories open_times eligibilities associated_sites type]
       # Define attributes used to build an Algolia record
       add_attribute :_geoloc do
-        if addresses.blank?
-          nil
-        else
+        if addresses.any?
```

Comment:
I'm wondering if we want to replace `blank` with `any` (and `empty`) in the service.rb file below. Apparently, `blank` can be called on `nil` values and on any object actually; whereas `any` and `empty` will raise a `nomethod` exception if called on `nil`. A `resource` and `service` should always at least have an empty addresses array (I believe), but I think it might be better to be safe and stick with `blank`.

Reply:
Ah I had changed it because `any?` gave me better results when I tested in `irb`. But it looks like `blank?` is an active record method or something, because it works the way you're describing when I test in the rails console. Happy to change it back.

Reply:
You can also use the [safe navigation operator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) (`&.`) to safely attempt to access methods on objects that might be `nil`. `addresses&.any?` will return `nil` if `addresses` is `nil` but will otherwise invoke `any?` on `addresses` when its not `nil`.

Reply:
I think using the `safe navigation operator` and leaving your changes would be a good way to go. Thanks for the info, Richard.